library/modules/Gui.cpp

```diff
@@ -948,16 +1011,66 @@ df::unit *Gui::getAnyUnit(df::viewscreen *top)
 
         return vector_get(world->units.active, *ui_selected_unit);
     }
+    case ZonesPitInfo: // (i) zone -> (P)it
+    case ZonesPenInfo: // (i) zone -> pe(N)
+    {
+        if (!ui_building_assign_units || !ui_building_item_cursor)
+            return NULL;
+
+        return vector_get(*ui_building_assign_units, *ui_building_item_cursor);
+    }
+    case Burrows:
+    {
+        if (ui->burrows.in_add_units_mode)
+            return vector_get(ui->burrows.list_units, ui->burrows.unit_cursor_pos);
+
+        return NULL;
+    }
+    case QueryBuilding:
+    {
+        df::building *building = getAnyBuilding(top);
+        if (building->getType() == df::building_type::Cage)
+        {
+            auto *cage = static_cast<df::building_cagest*>(building);
+            if (*ui_building_item_cursor < cage->assigned_units.size())
+                return df::unit::find(cage->assigned_units[*ui_building_item_cursor]);
+        }
+        return NULL;
+    }
     case LookAround:
     {
         if (!ui_look_list || !ui_look_cursor)
             return NULL;
 
         auto item = vector_get(ui_look_list->items, *ui_look_cursor);
-        if (item && item->type == df::ui_look_list::T_items::Unit)
-            return item->unit;
-        else
-            return NULL;
+        if (item) {
+            if (item->type == df::ui_look_list::T_items::Unit)
+                return item->unit;
+            else if (item->type == df::ui_look_list::T_items::Item)
+            {
+                if (item->item->getType() == df::item_type::CORPSE)
+                {
+                    // loo(k) at corpse
+                    auto *corpse = static_cast<df::item_corpsest*>(item->item);
+                    return df::unit::find(corpse->unit_id);
+                }
+                else if (item->item->getType() == df::item_type::CORPSEPIECE)
+                {
+                    // loo(k) at corpse piece
+                    auto *corpsepiece = static_cast<df::item_corpsepiecest*>(item->item);
+                    return df::unit::find(corpsepiece->unit_id);
+                }
+            }
+            else if (item->type == df::ui_look_list::T_items::Spatter)
+            {
+                // loo(k) at blood/ichor/.. spatter with a name
+                MaterialInfo mat;
+                if (mat.decode(item->spatter_mat_type, item->spatter_mat_index) && mat.figure)
+                    return df::unit::find(mat.figure->unit_id);
```

Comment:
You mentioned we shouldn't implement the find unit in blood spatter. I assume you meant not to implement find unit for a general race spatter. This spatter lookup will always find a single unit.

Reply:
It seems a bit weird to me for getSelectedUnit() to return a unit when no unit is selected, but it'll do for now.